jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/coder/stream/LogFileToAsn1CodecPublisher.java

```diff
@@ -69,23 +65,32 @@ public void publish(BufferedInputStream bis, String fileName, ImporterFileType f
    }
 
    private void publish(XmlUtils xmlUtils) throws JsonProcessingException {
-      OdeAsn1Payload payload = new OdeAsn1Payload(fileParser.getPayload());
-      OdeAsn1Metadata metadata = new OdeAsn1Metadata(payload);
-      metadata.getSerialId().setBundleId(bundleId.get()).addRecordId(1);
-      OdeLogMetadataCreatorHelper.updateLogMetadata(metadata, fileParser);
-      if (fileParser instanceof TimLogFileParser) {
-         ReceivedMessageDetails receivedMsgDetails = TimDecoderHelper.buildReceivedMessageDetails((TimLogFileParser) fileParser);
-         metadata.setReceivedMessageDetails(receivedMsgDetails);
-      }
-      
-      Asn1Encoding msgEncoding = new Asn1Encoding("root", "Ieee1609Dot2Data", EncodingRule.COER);
-      Asn1Encoding unsecuredDataEncoding = new Asn1Encoding("unsecuredData", "MessageFrame",
-            EncodingRule.UPER);
-      metadata.addEncoding(msgEncoding).addEncoding(unsecuredDataEncoding);
-      OdeAsn1Data asn1Data = new OdeAsn1Data(metadata, payload);
 
-      publisher.publish(xmlUtils.toXml(asn1Data),
-         publisher.getOdeProperties().getKafkaTopicAsn1DecoderInput());
+      if (fileParser instanceof DriverAlertFileParser){
+         logger.debug("Publishing a driver alert.");
+         publisher.publish(JsonUtils.toJson(fileParser, false),
```

Comment:
Architecturally, the ODE should always be sending out data in the consistent `{metadata:{}, payload:{}}` format. So I would recommend creating a `OdeDriverAlertPayload` and `OdeDriverAlertMetadata` and build it simliar to how it's done in the `else` part  and add `else` part to the existing `if (fileParser instanceof TimLogFileParser)`.

Reply:
So for the data contained in Driver Alerts, there isn't really a defined payload vs metadata fields. Potentially we could populate fields that that make sense as metadata and put the alert as the payload, but the alerts are just a short code. Thoughts?

Reply:
True, but nevertheless, some metadata fields are pretty standard such as `odeReceivedAt`, `serialId`, etc. We should also try to populate `recordGeneratedAt` and `recordGeneratedBy` if at all possible. Perhaps we could ask WYDOT to include a timestamp int he log file just like they do for BSMs. If can't do that, we should leave `recordGeneratedAt` blank but we could still populate `recordGeneratedBy`. The important thing is the consistency and uniformity of the ODE interface schema, architecturally speaking.

Reply:
Agreed, we can make those edits. Got a preference as to where `alert` should go?

Reply:
The `alert` itself will need to be the one and only element of a `OdeDriverAlert` class in `us.dot.its.jpo.ode.model` package extending `OdeObject`. `OdeDriverAlertPayload` and `OdeDriverAlertMetadata` should extend `OdeMsgPayload` and `OdeLogMetadata`, respectively. `OdeDriverAlert` is then set as the `data` element of `OdeDriverAlertPayload`. Model them after `OdeBsmPayload` and `OdeBsmMetadata`. Doing so will populate much of the common fields automatically if `OdeDriverAlertMetadata` is instantiated using the constructor that accepts a `OdeDriverAlertPayload`. So you will instantiate the payload first and then the metadata object given the payload.